### ISSUE TITLE
Select over disabled dates

### ISSUE THREAD

**Issue by iuliux** [Adphorus/react-date-range#337]
## Subject of the issue
A disabled date acts as a brickwall for any selection.
 
## [BUG] Bug Reproduce Steps
Let the `8th` be a disabled date.
I start a selection on the `4th`. The most this selection ca go is up to the `7th`. Even if I click the endDate on the `14th`, I still end up with a `4th - 7th` selection.

![Feb-04-2020 12-30-29 PM](https://user-images.githubusercontent.com/1446677/73737425-4bea7f80-474b-11ea-8c9d-1f8a4327aab3.gif)

## [BUG] Expected behaviour
If I select dates programatically, I will get the whole selection, with disabled dates marked as such.

<img width="330" alt="Screenshot 2020-02-04 at 12 31 27 PM" src="https://user-images.githubusercontent.com/1446677/73737520-74727980-474b-11ea-80dd-e3226503b596.png">

## Environment
Package Version: 1.0.0-beta2
React version: 16.12.0
Browser: Chrome

**Comment by Falkyouall** [comment]
Yes i ran into exactly the same issue and it would be nice to have a resolution for this.
